jstests/ssl/x509/mkcert.py

```diff
@@ -479,6 +482,35 @@ def process_client_multivalue_rdn(cert):
     os.remove(rsa)
     os.remove(pem)
 
+def convert_ecdsa_key(ec_key_file, pkcs8_key_file):
+    """
+    Convert ECDSA key with explicit text header into a PEM-encoded PKCS#8 object
+    :param ec_key_file: name of file containing the PEM-encoded ECDSA key with explicit text header
+    :param pkcs8_key_file: name of file to contain PEM-encoded PKCS#8 object describing the ECDSA key
+    """
+    # wrap the PEM-encoded EC key with explicit text header into a PEM-encoded PKCS#8 object
+    pkcs8args = ['openssl', 'pkcs8', '-topk8', '-nocrypt', '-in', ec_key_file, '-out', pkcs8_key_file]
+    subprocess.check_call(pkcs8args)
+
+def process_ecdsa_cert(cert, pem, key, dest, filename, split_pem=True):
+    """Convert the ECDSA key and write the public/private key pair key into .pem file, optionally writing the
+     public key to a .crt file and private e key to a .key file"""
+
+    # copy the public key to a .crt file
+    cert_filename = make_filename({'name': f"{filename}.crt"})
```

Comment:
I'm slightly uncomfortable with creating (possibly) temporary files in the working tree.  Could we make these `tempfile`s then conditionally move them into place if it turns out we do want to keep them?